Sources/RxMoya/MoyaProvider+Rx.swift

```diff
@@ -15,20 +15,8 @@ public extension Reactive where Base: MoyaProviderType {
     ///   - callbackQueue: Callback queue. If nil - queue from provider initializer will be used.
     /// - Returns: Single response object.
     public func request(_ token: Base.Target, callbackQueue: DispatchQueue? = nil) -> Single<Response> {
-        return base.rxRequest(token, callbackQueue: callbackQueue)
-    }
-
-    /// Designated request-making method with progress.
-    public func requestWithProgress(_ token: Base.Target, callbackQueue: DispatchQueue? = nil) -> Observable<ProgressResponse> {
-        return base.rxRequestWithProgress(token, callbackQueue: callbackQueue)
-    }
-}
-
-internal extension MoyaProviderType {
-
-    internal func rxRequest(_ token: Target, callbackQueue: DispatchQueue? = nil) -> Single<Response> {
-        return Single.create { [weak self] single in
-            let cancellableToken = self?.request(token, callbackQueue: callbackQueue, progress: nil) { result in
+        return Single.create { [weak base] single in
```

Comment:
Duh 😐 I just had to change `self` to `base`. Now I feel stupid 😝

Reply:
Don't be! Both in `RxSwift` and `ReactiveSwift` there is this struct called `ReactiveBase` or whatever, that is nothing more than the object that hold `rx` namespace and the object itself. E.g. when you have `provider.rx`, the `base` for it is a `MoyaProvider` object, `provider`.

Reply:
Yeah, I see now. `self` was referencing the `ReactiveBase` and `base` references the wrapped value. 🤦‍♂️ Thanks 👍

Reply:
I remember seeing this and being super confused. This is just a hack in the language to get nice namespaced Observables